documentation/modules/deploying/proc-deploy-migrate-kraft.adoc

```diff
@@ -0,0 +1,257 @@
+* `KRaftDualWriting` is the state when the migration has ended and the cluster is working as a KRaft cluster. 
+Metadata is managed in Kafka but also replicated in ZooKeeper, so you can xref:proc-deploy-migrate-kraft-rollback-{context}[roll back from this point].
+--
++
+The migration status is also represented in the `status.kafkaMetadataState` property of the `Kafka` resource. 
+
+. Enable KRaft in the `Kafka` resource configuration by setting the `strimzi.io/kraft` annotation to `enabled`:
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: enabled
+----
++
+.Enabling KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: enabled 
+# ...
+----
++
+WARNING: Rollback cannot be performed after enabling KRaft.
+
+. Check the status of the move to full KRaft mode:
++
+[source,shell]
+----
+kubectl get kafka my-cluster -n my-project -w
+----
++
+.Updates to the metadata state
+[source,shell]
+----
+NAME        ...  METADATA STATE        WARNINGS
+my-cluster  ...  Zookeeper
+my-cluster  ...  KRaftMigration
+my-cluster  ...  KRaftDualWriting
+my-cluster  ...  KRaftPostMigration
+my-cluster  ...  KRaft                 True
+----
++
+`KRaftPostMigration` is the state (after the brokers have rolled) when KRaft mode is enabled and there is no ZooKeeper involvement, though ZooKeeper pods are still running.
+`KRaft` is the final state (after the controllers have rolled) when the KRaft migration has finalized.
++
+The warning (`True`) relates to ZooKeeper configuration being present in the `Kafka` resource configuration.
++
+. Clean up the deployment to remove Zookeeper configuration and resources.
+
+.. Remove `inter.broker.protocol.version`, `log.message.format.version`, and all `spec.zookeeper` configuration properties from the `Kafka` resource.
+.. Apply the changes to the `Kafka` resource configuration.
+.. Delete any resources related to the ZooKeeper deployment:
++
+[source,shell]
+----
+kubectl delete <resource_type> <resource_name> -n my-project
+----
++
+For a list of resources created for ZooKeeper, see xref:ref-list-of-kafka-cluster-resources-str[].
+
+[id='proc-deploy-migrate-kraft-rollback-{context}']
+.Performing a rollback on the migration
+
+Before the migration is finalized by enabling KRaft in the 'Kafka' resource, you can perform a rollback operation as follows:
+
+. Apply the `strimzi.io/kraft: rollback` annotation to the `Kafka` resource to roll back the brokers and controllers.
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: rollback
+----
++
+.Rolling back KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: rollback 
+# ...
+----
+
+. Delete the controllers node pool:
```

Comment:
before this paragraph we should make clear that applying the "rollback" annotation is about rolling the brokers (to be connected to ZooKeeper again) and the state machine comes back to `KRaftDualWriting`.
When in this state, you can delete the controllers.

NOTE: actually from this state, the user could change their mind again and applying the "migration" flag to move again forward with migration.